modules/mono/glue/GodotSharp/GodotSharp/Core/Transform3D.cs

```diff
@@ -231,11 +263,21 @@ private void SetLookAt(Vector3 eye, Vector3 target, Vector3 up)
         }
 
         /// <summary>
-        /// Translates the transform by the given <paramref name="offset"/>,
-        /// relative to the transform's basis vectors.
-        ///
-        /// Unlike <see cref="Rotated"/> and <see cref="Scaled"/>,
-        /// this does not use matrix multiplication.
+        /// Translates the transform by the given <paramref name="offset"/>.
+        /// The operation is done in the parent/global frame, equivalent to
+        /// multiplying the matrix from the left.
+        /// </summary>
+        /// <param name="offset">The offset to translate by.</param>
+        /// <returns>The translated matrix.</returns>
+        public Transform3D Translated(Vector3 offset)
+        {
+            return new Transform3D(basis, origin + offset);
+        }
+
+        /// <summary>
+        /// Translates the transform by the given <paramref name="offset"/>.
+        /// The operation is done in the local frame, equivalent to
+        /// multiplying the matrix from the right.
         /// </summary>
         /// <param name="offset">The offset to translate by.</param>
         /// <returns>The translated matrix.</returns>
```

Comment:
I'm actually a bit surprised by the implementation of `TransledLocal` (a few lines down, I can only comment here). I'm not super familiar with C#, but doesn't the implementation mutate `this` because it writes to `origin`. Since this API is supposed to be an immutable API, I'm wondering if/why it also mutates. It feels like it should be just `+` instead of `+=` or even just `origin + basis.Xform(offset)` instead of manually constructing the three vector elements.

@aaronfranke What's your opinion on that?

Reply:
@bluenote10 You're right, that's a critical bug that needs fixing.

This bug may have been hidden from us before because if you write `myNode.Transform.Translated(offset)` then C# will operate on a *copy* of the transform. But, if you are working with a local variable, this bug would show up.

Reply:
Needs a rebase after that fix.

Reply:
@akien-mga Sure, done! Just to avoid confusion: The conflict was actually a result of this commit instead https://github.com/godotengine/godot/commit/a4ad1dfa517cf995f8aa5b913ac9726abca620f7 

@aaronfranke's fix was in Transfrom3D.cs not the Transform2D.cs, so I was confused at first...